app/login/page.tsx

```diff
@@ -19,86 +21,42 @@ const LoginPage: React.FC = () => {
-    const router = useRouter();
     const errorRef = useRef<HTMLDivElement>(null);
+    const { csrfToken } = useCsrf();
+    const router = useRouter();
     const { setIsAuthenticated } = useAuthContext();
 
-
     const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
         setUser({ ...user, [event.target.name]: event.target.value });
     };
 
-    const saveTokenInCookie = async (token: string) => {
-
-        //Set the cookie. Needs to be set on server side in order to have security configs like httpOnly
-        const result = await fetch('/api/setCookie', {
-            method: 'POST',
-            headers: {
-                'Content-Type': 'application/json',
-            },
-            body: JSON.stringify({ token })
-        })
-
-        const data = await result.json();
-        if (data.status !== 200) {
-            logECS('error', data.message, {
-                url: { path: '/apollo-signin' }
-            });
-        } else {
-            logECS('info', data.message, {
-                url: { path: '/apollo-signin' }
-            });
-        }
-    }
-
-    const handleResponse = async (response: Response) => {
-        const { token, message } = await response.json();
-
-        if (response.status === 200) {
-            await saveTokenInCookie(token)
-            setIsAuthenticated(true);
-            router.push('/') //redirect to home page
-        } else if (response.status === 401) {
-            if (message) {
-                setErrors(prevErrors => [...prevErrors, message])
-            }
-        } else if (response.status === 500) {
-            logECS('error', 'Internal server error', {
-                url: { path: '/apollo-signin' }
-            });
-            router.push('/500-error');
-        } else {
-            setErrors(['An unexpected error occurred. Please try again.'])
-        }
-    };
-
     const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
         event.preventDefault();
         setLoading(true);
         setErrors([]); // Clear previous errors
 
-        if (attempts > 5) {
-            const remainingTime = lockoutTime ? Math.max(lockoutTime - Date.now(), 0) : 0;
-            const minutesLeft = Math.ceil(remainingTime / 60000);
-            const remainingMinutesText = minutesLeft < 2 ? 'minute' : 'minutes'
-            setErrors([`Too many attempts. Please try again later${remainingTime > 0 ? ` in ${Math.ceil(remainingTime / 60000)} ${remainingMinutesText}.` : '.'}`]);
-            setLoading(false);
-            return;
-        }
-
-        /* eslint-disable @typescript-eslint/no-explicit-any */
-        try {
-            const response = await fetch(`${process.env.NEXT_PUBLIC_SERVER_ENDPOINT}/apollo-signin`, {
+        const loginRequest = async (token: string | null) => {
+            return await fetch(`${process.env.NEXT_PUBLIC_SERVER_ENDPOINT}/apollo-signin`, {
```

Comment:
These fetch requests can probably be moved to a separate function. I didn't bother doing it for now, since I know there is a ticket to create the /login and /signup pages. My focus was just to get them working with the backend for this ticket.